rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,396 @@
+A peer can issue an IDONTWANT only after it has received the entire message. 
+However, a large message transmission may take several hundred milliseconds to complete. 
+During this time, many other mesh members may start relaying the same message. 
+
+Therefore, the probability of simultaneously receiving the same message from multiple senders increases with the message size, 
+significantly compromising the effectiveness of IDONTWANT messages.
+
+Introducing a handshake phase for large message transmissions, 
+where the sender shares msgID and length, can provide valuable insight into ongoing message receptions. 
+
+If the receiver is already receiving this message from another sender, 
+it can request to defer this transmission. 
+
+An IDONTWANT from the receiver will indicate successful message reception. 
+Otherwise, the sender can initiate transmission after a specific wait interval. 
+
+To simplify this process, a peer can send a brief IMReceiving message (for all ongoing large message transmissions), 
+urging all mesh peers to hold off on sending the same message [[15](https://forum.vac.dev/t/large-message-handling-idontwant-imreceiving/281),[16](https://forum.vac.dev/t/idontwant-message-impact/283)]. 
+If no IDONTWANT is received from the peer within a specific time, 
+mesh peers can resume this transmission. 
+This approach can boost IDONTWANT benefits by also considering ongoing transmissions for large messages. 
+
+While IMReceiving messages can bring about substantial improvements in terms of latency and bandwidth utilization, 
+it's crucial to be aware of the potential risks. 
+
+This approach could be exploited by a malicious user to disrupt message transmission, 
+either by never completing a message or by intentionally sending a message at an extremely slow rate to numerous peers. 
+This could ultimately result in network-wide slow message propagation. 
+
+However, with careful calibration of the deferring interval and peer scoring, these risks can be mitigated.
+
+#### d. IDONTWANT Message with Reduced Forwarding
+It is common for slow peers to pile up outgoing message queues, 
+especially for large message transfers.
+
+This results in a significant queuing delay for outgoing messages. 
+Reduced message forwarding can help decrease the workload of slower peers. 
+
+On receiving a message longer than the specified threshold, 
+a slow peer can relay it to only $K \in D$ peers and sends an IDONTWANT message to all the peers in $D$.
+This approach helps save peer bandwidth by minimizing redundant transmissions. 
+
+At the same time, The IDONTWANT message can play a crucial role in enabling the remaining peers to fetch missing messages using IWANT requests. 
+
+This method is particularly effective in minimizing network-wide message dissemination latency, 
+allowing the remaining peers to retrieve the message almost immediately 
+(approximately in $RTT + \tau$ time) without waiting for the heartbeat (gossip) interval. 

```

Comment:
Could you explain the reasoning behind  RTT+ \tau?

Reply:
Yes it should be RTT,,, tau is consumed in both cases. I have updated the section

c94893bb9d18129a6973a7963b316b3bee4d24f1